pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -412,17 +434,38 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 		need := uint64(config.FixedResources.Memory) << 10
 		if remaining < need {
 			var errStr string
+			var entities []*types.ErrorEntity
+			errSeverity := types.ErrorSeverityWarning
+			retryCondition := "Retry will be triggered when apps will shutdown: "
 			if remaining+halting < need {
 				errStr = fmt.Sprintf("Remaining memory bytes %d app instance needs %d",
 					remaining, need)
+				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
+					status := st.(types.AppInstanceStatus)
+					entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
+					retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
+				}
 			} else {
 				errStr = fmt.Sprintf("App instance needs %d bytes but only have %d; waiting for halting app instances to free up %d bytes",
 					need, remaining, halting)
+				errSeverity = types.ErrorSeverityNotice
+				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
+					status := st.(types.AppInstanceStatus)
+					if status.Activated || status.ActivateInprogress {
+						entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
+						retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
```

Comment:
Good enough for now, but maybe the UI will not need this string concan and will use the enties  instead